### PR TITLE
http2: simplifies inbound frame tracking

### PR DESCRIPTION
**Pull request by birenroy**
Although ProtocolConstraints::trackInboundFrames() accepts almost the entire HTTP/2 frame header, all it really needs to know is the frame type, whether the frame is semantically empty, and whether the frame indicates the end of stream.

Commit Message: http2: simplifies inbound frame tracking
Additional Description:
Risk Level: low
Testing: ran unit and integration tests locally
Docs Changes:
Release Notes:
Platform Specific Features:

### CODE REVIEW

**Comment by birenroy** [comment]
/assign @diannahu 
/assign @RyanTheOptimist 

If you want a preview of the diffs once https://github.com/envoyproxy/envoy/pull/32078 is merged, try this link: https://github.com/birenroy/envoy/compare/merge-connectionimpl-trackinboundframes...birenroy:envoy:simplify-inbound-frame-tracking?expand=1

**Review by RyanTheOptimist — COMMENTED** [review]
Nice! looks great! I'll approve once the other PR lands, but this looks nice!

**Review by RyanTheOptimist — APPROVED** [review]

File: source/common/http/http2/protocol_constraints.cc
Comment (diannahu) [inline source/common/http/http2/protocol_constraints.cc:68]: double-checking: NGHTTP2_CONTINUATION is not expected to happen here also IIUC, yeah? (If it were, would ask to include in codec_impl.cc's end_stream condition, but because it isn't IIUC, we're good.)

**Review by diannahu — APPROVED** [review]
Nice clean-up/simplification!

File: source/common/http/http2/protocol_constraints.cc
Comment (birenroy) [reply source/common/http/http2/protocol_constraints.cc:68]: Yes, that should be true. From a frame-tracking perspective, [HEADERS + CONTINUATION sequence] are all considered one frame.

**Comment by birenroy** [comment]
CI passed, this is ready for merge.

**Pull request merged**
